### DISCUSSION TITLE
Flatten data using N:N relationship

### DISCUSSION THREAD

**Discussion by SimGuyCox** [rappen/FetchXMLBuilder#728] [Q&A]
Curious if anyone has a clever way of using fetchxml to flatten out data for an N:N relationship. I can achieve it with a 1:N but not an N:N

Desired outcome would look like below (note the link entity and alias)

![image](https://user-images.githubusercontent.com/107057393/172411033-f52bab8e-0d6d-4937-86d3-74f65b2c604c.png)

The above eliminates multiple duplicate rows with the following fetchxml against 1:N relationship.

![image](https://user-images.githubusercontent.com/107057393/172411579-4d51af2b-ebcb-4823-bc82-682e7c47f3da.png)

But I cannot achieve the same thing when using an OOTB N:N Relationship as it duplicates records for each occurrence of related records.

Thanks
Guy

**Comment by rappen** [comment]
I think this has no answer... 😕
It's just as hard as for the 1:N relationship.
You could compare it to Account 1:N Contact, you can never join with Contact to the base entity Account. It will show duplicated Accounts for each Contact that Account has.
Using N:1 is a not easier, like Contact N:1 Account.

N:N is actually N:1:N, even if you filter it like your query, technically you are locked into the relationships...

(I may have misunderstood everything - if so, please correct me!)
